src/error_log.cpp

```diff
@@ -81,5 +81,193 @@ void FFDCFile::removeFFDCFile()
+    }
+    return std::nullopt;
+}
+
+std::optional<std::vector<std::string>>
+    FFDCFileSet::getJournalTraces(const std::string& fieldValue,
+                                  const unsigned int maxTraces)
+{
+    // systemd journal context
+    sd_journal* journal;
+    int ret = sd_journal_open(&journal, SD_JOURNAL_LOCAL_ONLY);
+    if (0 == ret)
+    {
+        std::vector<std::string> traces;
+
+        // To get the latest entries, iterating the journal in reverse order
+        SD_JOURNAL_FOREACH_BACKWARDS(journal)
+        {
+            auto retValue = getTraceFieldValue(journal, "SYSLOG_IDENTIFIER");
+            // Get the next traces if not the fieldValue
+            if (!retValue.has_value() || *retValue != fieldValue)
+            {
+                continue;
+            }
+
+            // Get timeStamp
+            std::string timeStamp;
+            uint64_t usec{0};
+            if (0 == sd_journal_get_realtime_usec(journal, &usec))
+            {
+                // Converting realtime microseconds to human readable format
+                // E.g. Nov 06 2025 14:23:45
+                std::time_t timeInSec = static_cast<time_t>(usec / 1000000);
+                std::stringstream ss;
+                ss << std::put_time(std::localtime(&timeInSec),
+                                    "%b %d %Y %H:%M:%S");
+                timeStamp = ss.str();
+            }
+
+            // Get Process PID
+            std::string pid;
+            retValue = getTraceFieldValue(journal, "_PID");
+            if (retValue.has_value())
+            {
+                pid = *retValue;
+            }
+
+            // Get Message
+            std::string message;
+            retValue = getTraceFieldValue(journal, "MESSAGE");
+            if (retValue.has_value())
+            {
+                message = *retValue;
+            }
+
+            // Format -> Timestamp : ProcessName[ProcessPID] : Message
+            std::stringstream trace;
+            trace << timeStamp << " : " << fieldValue << "[" << pid
+                  << "] : " << message;
+            traces.push_back(trace.str());
+
+            if (traces.size() == maxTraces)
+            {
+                break;
+            }
+        }
+
+        sd_journal_close(journal);
+
+        if (!traces.empty())
+        {
+            return traces;
+        }
+        else
+        {
+            lg2::info("No journal traces found for {FIELD_VALUE}",
+                      "FIELD_VALUE", fieldValue);
+        }
+    }
+    else
+    {
+        lg2::error(
+            "Failed to get systemd journal traces for {FIELD_VALUE}, error code:"
+            "{ERROR_CODE}, error message: {ERROR_MSG}",
+            "FIELD_VALUE", fieldValue, "ERROR", ret, "ERROR_MSG",
+            strerror(ret));
+    }
+    return std::nullopt;
+}
+
+void FFDCFileSet::createFFDCFilesForTraces()
+{
+    // Not including rsyncd to prevent overflow of traces
```

Comment:
I think it bit confusing.I would add comment like below

"rsyncd traces are disabled to avoid trace overflow in the journal. Since the data-sync trace already captures the rsync error details, rsyncd traces are not included in the FFDC."